polaris-core/src/main/java/org/apache/polaris/core/auth/PolarisAuthorizer.java

```diff
@@ -514,12 +517,12 @@ public void authorizeOrThrow(
           "Principal '%s' is not authorized for op %s due to PRINCIPAL_CREDENTIAL_ROTATION_REQUIRED_STATE",
           authenticatedPrincipal.getName(), authzOp);
     } else if (!isAuthorized(
-        authenticatedPrincipal, activatedGranteeIds, authzOp, targets, secondaries)) {
+        authenticatedPrincipal, activatedEntities, authzOp, targets, secondaries)) {
       throw new ForbiddenException(
-          "Principal '%s' with activated PrincipalRoles '%s' and activated ids '%s' is not authorized for op %s",
+          "Principal '%s' with activated PrincipalRoles '%s' and activated grants via '%s' is not authorized for op %s",
```

Comment:
Minor: Not sure if this is clearer? `grants via` -> `grant on`

Reply:
I think what this message is trying to say is that you have grants _through_ / _via_ these roles. So you may have a grant on table X through your principal role A and catalog role B but you fail when you try to read table Y. This message will say `...and activated grants via [A, B]`

Reply:
I guess you also have grants _on_ A and B but they are not privilege grants, they are role grants.

Reply:
My bad. I thought they table/namespace entities.

Reply:
Ah no they are roles. But I am still not sure if this verbiage is totally correct, maybe you are right

Reply:
The `via` makes sense to me if they are roles.